### PR TITLE
thiccc: set app=aws-operator and version labels

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/7650

This PR set app and version labels for service, deployment, and pod resources.

### CODE REVIEW

File: helm/aws-operator-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ metadata:
   namespace: {{ .Values.namespace }}
   labels:
     app: aws-operator
+    version: {{ .Chart.Version }}
 spec:
```
Comment (kopiczko) [inline helm/aws-operator-chart/templates/deployment.yaml]: Don't we have selector in the deployment?

Comment (TheoBrigitte) [reply helm/aws-operator-chart/templates/deployment.yaml]: There are no selector in the deployment, this is just to align deployment and pod labels.

**Comment by TheoBrigitte** [comment]
Tested on gauss

```
$ k get all -A -lapp=aws-operator,version=1.0.0-a13f9abeca4d2e824c9dd70d782e582bbe1b2b41 --show-labels
NAMESPACE    NAME                               READY   STATUS    RESTARTS   AGE   LABELS
giantswarm   pod/aws-operator-db9b8d476-mjrwp   1/1     Running   0          41s   app=aws-operator,pod-template-hash=db9b8d476,version=1.0.0-a13f9abeca4d2e824c9dd70d782e582bbe1b2b41

NAMESPACE    NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE   LABELS
giantswarm   service/aws-operator   ClusterIP   172.31.235.90   <none>        8000/TCP   96d   app=aws-operator,version=1.0.0-a13f9abeca4d2e824c9dd70d782e582bbe1b2b41

NAMESPACE    NAME                           READY   UP-TO-DATE   AVAILABLE   AGE   LABELS
giantswarm   deployment.apps/aws-operator   1/1     1            1           96d   app=aws-operator,version=1.0.0-a13f9abeca4d2e824c9dd70d782e582bbe1b2b41

NAMESPACE    NAME                                     DESIRED   CURRENT   READY   AGE   LABELS
giantswarm   replicaset.apps/aws-operator-db9b8d476   1         1         1       41s   app=aws-operator,pod-template-hash=db9b8d476,version=1.0.0-a13f9abeca4d2e824c9dd70d782e582bbe1b2b41
```

File: helm/aws-operator-chart/templates/deployment.yaml (hunk 1 above)
Comment (kopiczko) [reply helm/aws-operator-chart/templates/deployment.yaml]: Do you know how it works? Does it take its own labels and create selector out of that by default?

Comment (TheoBrigitte) [reply helm/aws-operator-chart/templates/deployment.yaml]: When deployment selector is empty, it takes the pod labels by default. https://godoc.org/k8s.io/api/extensions/v1beta1#ReplicaSetSpec

Selector must be specified, starting from k8s 1.15 in api version `apps/v1beta2`, but we are still using `extensions/v1beta1` here. https://gigantic.slack.com/archives/C12SG2929/p1574337314118300

All good, let's move on!

Comment (kopiczko) [reply helm/aws-operator-chart/templates/deployment.yaml]: 👍

**Pull request merged**